examples/systemcalls/systemcalls.c

```diff
@@ -58,6 +73,69 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
+    pid_t pid_ch;
+
+    //Create a new child process
+    pid_ch = fork(); //fork() returns pid
+    if(pid_ch == -1)
+    {
+        perror("Fork error");
+        return false;
+    }
+
+    else if(pid_ch == 0)
+    {
+        //execv which replaces the existing parent process
+        int ret;
+        ret = execv(command[0], command);
+
+        if(ret == -1)
+        {
+            perror("execv returned error");
+        }
+
+        printf("Error executing command execv. Error type: %d\n", errno);
+        exit(1);
+    }
+
+    else
+    {
+        int status;
+        pid_t pid_w;
+        pid_w = waitpid(pid_ch, &status, 0); //Using waitpid to pass child process pid
+        
+        if(pid_w == -1)
+        {
+            printf("Error executing waitpid. Error type: %d\n", errno);
+            perror("Waitpid error");
+            return false;
```

Comment:
Missing cleanup in these code paths.  Also see discussion in class about avoiding duplication between these two very similiar functions.